src/shared/json-queue.c

```diff
@@ -108,45 +108,56 @@ void jqueue_close(file_queue * queue) {
  * @param queue pointer to the file_queue struct
  * @post The flag variable may be set to CRALERT_READ_FAILED if the read operation got no data.
  * @post The read position is restored if failed to get a JSON object.
- * @retval NULL No data read or could not get a valid JSON object. Pointer to the JSON object otherwise.
+ * @retval NULL No data read or could not get a valid JSON object or read overlong alert. Pointer to the JSON object otherwise.
  */
 cJSON * jqueue_parse_json(file_queue * queue) {
     cJSON * object = NULL;
     char buffer[OS_MAXSTR + 1];
+    int64_t initial_pos;
     int64_t current_pos;
+    int64_t offset;
     const char * jsonErrPtr;
     char * end;
 
-    current_pos = w_ftell(queue->fp);
+    initial_pos = w_ftell(queue->fp);
 
     if (fgets(buffer, OS_MAXSTR + 1, queue->fp)) {
 
         if (end = strchr(buffer, '\n'), end) {
             *end = '\0';
 
             if ((object = cJSON_ParseWithOpts(buffer, &jsonErrPtr, 0), object) && (*jsonErrPtr == '\0')) {
-                queue->read_attempts = 0;
                 return object;
             }
-        }
 
-        // The read JSON is invalid
-        if (object) {
+            // The read JSON is invalid
             cJSON_Delete(object);
-        }
 
-        queue->read_attempts++;
-        mdebug2("Invalid JSON alert read from '%s'. Remaining attempts: %d", queue->file_name, MAX_READ_ATTEMPTS - queue->read_attempts);
+            mwarn("Invalid JSON alert read from '%s': '%s'", queue->file_name, buffer);
+            return NULL;
+        }
 
-        if (queue->read_attempts < MAX_READ_ATTEMPTS) {
-            if (current_pos >= 0) {
-                if (fseek(queue->fp, current_pos, SEEK_SET) != 0) {
-                    queue->flags = CRALERT_READ_FAILED;
+        current_pos = initial_pos;
+        offset = w_ftell(queue->fp);
+        while ((offset-current_pos) == OS_MAXSTR) {
+            if (fgets(buffer, OS_MAXSTR + 1, queue->fp)) {
+                current_pos = offset;
```

Comment:
These two lines can be executed after the if from below.

Reply:
Done!